src/state_machine/coordinator/fire.rs

```diff
@@ -943,91 +943,101 @@ impl<Aggregator: AggregatorTrait> Coordinator<Aggregator> {
         packet: &Packet,
         signature_type: SignatureType,
     ) -> Result<(), Error> {
-        if let Message::SignatureShareResponse(sig_share_response) = &packet.msg {
-            if sig_share_response.dkg_id != self.current_dkg_id {
-                return Err(Error::BadDkgId(
-                    sig_share_response.dkg_id,
-                    self.current_dkg_id,
-                ));
-            }
-            if sig_share_response.sign_id != self.current_sign_id {
-                return Err(Error::BadSignId(
-                    sig_share_response.sign_id,
-                    self.current_sign_id,
-                ));
-            }
+        let Message::SignatureShareResponse(sig_share_response) = &packet.msg else {
+            return Ok(());
+        };
 
-            // check that the signer_id exists in the config
-            let signer_public_keys = &self.config.signer_public_keys;
-            if !signer_public_keys.contains_key(&sig_share_response.signer_id) {
-                warn!(signer_id = %sig_share_response.signer_id, "No public key in config");
-                return Ok(());
-            };
+        let response_info = self.message_nonces.entry(self.message.clone()).or_default();
```

Comment:
qq: why the insertion in `message_nonces` here? IIUC if we are in sigshares gather, we should have an entry populated in `gather_nonces`; if we insert here, the next `if` will fail (empty default `sign_wait_signer_ids`), so it doesn't seem we have any reason to insert and we could just `get` and err if none